src/map/src/combat/player.cpp

```diff
@@ -0,0 +1,77 @@
+#include "dataconsts.h"
+#include "cmapclient.h"
+#include "combat/player.h"
+#include "combat/combat.h"
+#include "entity_system.h"
+#include "components/basic_info.h"
+#include "components/stats.h"
+#include "components/computed_values.h"
+#include "cli_stat_add_req.h"
+#include "cli_toggle_move.h"
+#include "srv_stat_add_reply.h"
+#include "srv_toggle_move.h"
+#include <cmath>
+#include "utils/calculation.h"
+
+
+using namespace RoseCommon;
+using namespace Player;
+
+void Player::add_stat(EntitySystem& entitySystem, RoseCommon::Entity entity, const RoseCommon::Packet::CliStatAddReq& packet) {
+	auto& basicInfo = entitySystem.get_component<Component::BasicInfo>(entity);
+	auto& stats = entitySystem.get_component<Component::Stats>(entity);
+	uint8_t statId = packet.get_stat();
+	uint32_t neededStatPoints = 0;
+	uint16_t newStatValue = 0;
+	// Calculating the needed points for a stat raise
+	switch(statId)
+    {
+		case 0: neededStatPoints = (int)floor((float)stats.str / 5); break;
+		case 1: neededStatPoints = (int)floor((float)stats.dex / 5); break;
+		case 2: neededStatPoints = (int)floor((float)stats.int_ / 5); break;
+		case 3: neededStatPoints = (int)floor((float)stats.con / 5); break;
+		case 4: neededStatPoints = (int)floor((float)stats.charm / 5); break;
+		case 5: neededStatPoints = (int)floor((float)stats.sense / 5); break;
+	}
+
+	// Is there enough stat points AND stat is not maxed?
+	if (basicInfo.statPoints >= neededStatPoints && neededStatPoints < (MAX_STAT / 5)) {
+		switch(statId)
+        {
+			case 0: newStatValue = ++stats.str; break;
+			case 1: newStatValue = ++stats.dex; break;
+			case 2: newStatValue = ++stats.int_; break;
+			case 3: newStatValue = ++stats.con; break;
+			case 4: newStatValue = ++stats.charm; break;
+			case 5: newStatValue = ++stats.sense; break;
+		}
+		basicInfo.statPoints -= neededStatPoints;
+		// Updating HP & MP
+		Combat::updateStats(entitySystem, entity);
+		auto pStat = Packet::SrvStatAddReply::create(static_cast<Packet::SrvStatAddReply::Stat>(statId), newStatValue);
+		entitySystem.send_to(entity, pStat);
+	}
+
+}
+
+void Player::toggle_player_move(EntitySystem& entitySystem, RoseCommon::Entity entity, const RoseCommon::Packet::CliToggleMove& packet) {
+    auto logger = Core::CLog::GetLogger(Core::log_type::GENERAL).lock();
+
+	Packet::CliToggleMove::ToggleMove moveType = packet.get_type();
+	auto& computedValues = entitySystem.get_component<Component::ComputedValues>(entity);
+	if(moveType == 0) {
+		if (computedValues.moveMode == MoveMode::RUN) {
+			computedValues.moveMode = MoveMode::WALK;
+		} else if (computedValues.moveMode == MoveMode::WALK) {
+			computedValues.runSpeed = Calculations::get_runspeed(entitySystem, entity); //get real run speed
+			computedValues.moveMode = MoveMode::RUN;
+		}
+	}
+	entitySystem.save_character(entity);
```

Comment:
Couldn't this cause an issue?

Reply:
Not sure it is needed.
Not sure it can cause an issue either. Why would it?

Reply:
I just wanted @L3nn0x to comment on this.